tests/test_edge_cases.py

```diff
@@ -1149,7 +1149,7 @@ class Model(BaseModel):
     assert exc_info.value.errors() == [{'input': {}, 'loc': ('bar',), 'msg': 'Field required', 'type': 'missing'}]
 
 
-@pytest.mark.xfail(reason='working on V2 - validators')
+@pytest.mark.xfail(reason='items yielded by __get_validators__ are not inspected for valid signatures')
```

Comment:
Should we be inspecting them? I was thinking that the `@validator` versions would get inspected and support several signatures for brevity but the versions defined by `__get_validators__` would only support the exact pydantic-core signauture and be enforced only via typing. Reason being that `@validator` gets used a lot more frequently so it's nice to be succint and get more errors caught early but `__get_validators__` is more rare thus it's okay to force users to be more verbose.

Reply:
Not sure, but that _is_ the reason the test is failing 😄

Reply:
Yep next pass may be fixing any signatures that aren’t V2 compliant and deleting this test…